extensions/ql-vscode/src/data-extensions-editor/data-extensions-editor-view.ts

```diff
@@ -86,15 +91,36 @@ export class DataExtensionsEditorView extends AbstractWebview<
         await this.postMessage({ t: "addModeledMethods", modeledMethods });
       },
     );
+    this.externalApiUsages = [];
   }
 
   public async openView() {
     const panel = await this.getPanel();
     panel.reveal(undefined, true);
 
+    panel.onDidChangeViewState(async () => {
+      if (await this.isTheMostRecentlyActivePanel()) {
+        await this.updateModelDetailsPanelState(
+          this.externalApiUsages,
+          this.databaseItem,
+        );
+      }
+    });
+
     await this.waitForPanelLoaded();
   }
 
+  private async isTheMostRecentlyActivePanel(): Promise<boolean> {
```

Comment:
We could split this into two methods, so we call a different method when the view state changes vs when we just want to check if we're the most-recently active. But I also kinda like the simplicity and security of just having one method that checks and updates `DataExtensionsEditorView.mostRecentlyActivePanel`.

Reply:
I wouldn't expect a method called `isTheMostRecentlyActivePanel` to make any updates to the state. I think splitting this into two methods would make sense (one of which can be a getter).

Reply:
That is a very good point, that it's not clear from the name that it also updates state. I'll split it into two methods.